.openshift-ci/get_latest_helm_chart_versions.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env python3
+
+"""
+Looks up N previous releases and outputs a Helm chart version for the most recent patch for each found release.
+"""
+
+import json
+import logging
+import pathlib
+import re
+import subprocess
+import sys
+
+from collections import namedtuple
+
+this_script_dir = pathlib.Path(__file__).parent
+repo_root = this_script_dir.parent
+
+helm_repo_name = "temp-stackrox-oss-repo-should-not-see-me"
+
+add_repo_cmd = f"helm repo add {helm_repo_name} https://raw.githubusercontent.com/stackrox/helm-charts/main/opensource"
+update_repo_cmd = "helm repo update"
+search_cmd = f"helm search repo {helm_repo_name} --versions --output json"
+remove_repo_cmd = f"helm repo remove {helm_repo_name}"
+
+Version = namedtuple("Version", ["major", "minor", "patch"])
+
+# Here we call "release" (or Y-Stream) the first appearance of X.Y.0 version.
+Release = namedtuple("Release", ["major", "minor"])
+
+# Default value of N, the number of previous releases to look up.
+# The current release cadence is 9 weeks (sometimes extended but not reduced), i.e. 9*7=63 days.
+# The current support period is 6 months, i.e. at most 184 days.
+# Therefore, at most 3 releases will be in support at any given moment of time with the current cadence and support
+# period.
+num_releases_default = 3
+
+
+def main(argv):
+    logging.basicConfig(stream=sys.stderr, level=logging.DEBUG)
+    n = int(argv[1]) if len(argv) > 1 else num_releases_default
+    helm_versions = get_latest_helm_chart_versions("stackrox-secured-cluster-services", n)
+    logging.info(f"Helm chart versions for the latest {n} releases:")
+    print("\n".join(helm_versions))
+
+
+def get_latest_helm_chart_versions(chart_name, num_releases=num_releases_default):
+    add_helm_repo()
+    try:
+        update_helm_repo()
+        return __get_latest_helm_chart_versions(chart_name, num_releases)
+    finally:
+        remove_helm_repo()
+
+
+def __get_latest_helm_chart_versions(chart_name, num_releases):
+    charts = read_charts()
+    logging.info(f"Discovered total {len(charts)} charts")
+
+    filtered_charts = filter_charts_by_name(charts, chart_name)
+    logging.info(f"Found {len(filtered_charts)} charts with the given name {chart_name}")
+
+    latest_charts = get_latest_chart_for_each_release(filtered_charts)[:num_releases]
+    logging.debug(f"Identified these charts as {num_releases} latest: {latest_charts}")
+
+    return [c["version"] for c in latest_charts]
+
+
+def read_charts():
+    json_str = run_command(search_cmd, log_stdout=False)
+    charts_from_json = json.loads(json_str)
+
+    release_charts = [c for c in charts_from_json if is_release_version(c["app_version"])]
```

Comment:
1. Again, this was there already.
2. You're right, there are no other versions currently, but the code makes assumption that the version is according to this format and will break otherwise. Moreover, anything more custom will be like `1.2.3-blah` or `1.2.3+blah` which is **not** a release version according to SemVer. Since we need to test against release versions, we filter it this way.